lib/src/wrappers.dart

```diff
@@ -547,6 +567,24 @@ class MapKeySet<E> extends _DelegatingIterableBase<E>
 ///
 /// Effectively, the map will act as a kind of index for the set.
 class MapValueSet<K, V> extends _DelegatingIterableBase<V> implements Set<V> {
+  /// Adapts [source] to be a `MapKeySet<T>`.
+  ///
+  /// Any time the set would produce an element that is not a [T], the element
+  /// access will throw.
+  ///
+  /// Any time a [T] value is attempted stored into the adapted set, the store
+  /// will throw unless the value is also an instance of [S].
+  ///
+  /// If all accessed elements of [source] are actually instances of [T] and if
+  /// all elements stored in the returned  are actually instance of [S],
+  /// then the returned set can be used as a `MapKeySet<T>`.
+  static MapValueSet<K2, V2> castFrom<K, V, K2, V2>(MapValueSet<K, V> source) {
```

Comment:
Again, this class is just an implementation of `Set`, so just use `Set.castFrom`.

Reply:
Done.